src/pages/ItemList.js

```diff
@@ -74,38 +77,70 @@ function ItemList({ token }) {
     fetchDocs(token);
   };
 
+  useEffect(() => {
+    const filterList = (e) => {
```

Comment:
This function is super cool :)  Once I realized exactly how it worked, it just made me smile.  The whole thing y'all did is just such a nice solution!  I suppose you could get rid of the console.log but that is about it.